### ISSUE TITLE
Need testing on ManagedArray<const T>

### ISSUE THREAD

**Issue by DavidPoliakoff** [LLNL/CHAI#16]
```
#include <chai/ManagedArray.hpp>
int main(){
  chai::ManagedArray<const double> foo(50);
  const double* foo2 = (const double*) foo;
}
```

This will cause an error in the casting operator, as registerTouch is ill-defined:

```
ArrayManager.hpp:113:8: note: candidate function not viable: no known conversion from 'const double *' to 'void *' for 1st argument;
```

I'm writing workarounds, but it'd be nice for this to work.

**Comment by DavidPoliakoff** [comment]
Note: the fix turned out to be trivial, feel free to merge the PR or use it as a template

**Comment by davidbeckingsale** [comment]
Closed by #17
